code/network/multimsgs.cpp

```diff
@@ -8011,14 +8011,16 @@ static constexpr size_t animation_forced_bit = 1 << 1;
 static constexpr size_t animation_instant_bit = 1 << 2;
 static constexpr size_t animation_pause_bit = 1 << 3;
 
-void send_animation_triggered_packet(int animationId, int pmi, const animation::ModelAnimationDirection& direction, bool force, bool instant, bool pause, const int* /*time*/) {
+void send_animation_triggered_packet(unsigned int animationId, object* parent_object, ushort special_mode, const animation::ModelAnimationDirection& direction, bool force, bool instant, bool pause, const int* /*time*/) {
 	int packet_size;
+	ushort netsig_to_send = special_mode == 0 && parent_object != nullptr ? parent_object->net_signature : (ushort)0;
```

Comment:
I get the idea here, but I worry about it being too confusing or difficult to maintain in the future. It would be cleaner to just initialize it to zero and add an `if()` statement to conditionally assign the net_signature. That would make more sense to others looking at the code a few years down the line I think.

Also, if a non-null `parent_object` together with a non-zero `special_mode` shouldn't happen then it should either warn or error so we'll know to do something about it. And the `if()` statement would provide a good place for that.

Reply:
A non-zero object and a non-zero special_mode _could_ be valid, I can imagine. There's no case for it right now, but there could be special things that are technically objects, but should be synced differently in the future. Hence that possibility should be valid.
If both are zero, however, than this indeed needs an error, or probably really an Assert. It would currently just be discarded in the packet recieve method, but is technically never valid and should not occur.